sdk/core/src/request_options/next_marker.rs

```diff
@@ -23,6 +24,27 @@ impl NextMarker {
     pub fn as_str(&self) -> &str {
         &self.0
     }
+
+    pub fn append_to_url_query_as_continuation(&self, url: &mut url::Url) {
+        url.query_pairs_mut().append_pair("continuation", &self.0);
+    }
+
+    pub fn from_header_optional(headers: &http::HeaderMap) -> Result<Option<Self>, AzureError> {
+        let header_as_str = headers
+            .get("x-ms-continuation")
+            .map(|item| item.to_str())
+            .transpose()?;
+
+        Ok(if let Some(nm) = header_as_str {
```

Comment:
nit: `header_as_str.filter(|nm| nm.is_empty()).map(|nm| NextMarker::new(nm.to_owned()))`

Reply:
Nice! Copypasting it in! 👍

Reply:
Woops... it should be `!nm.is_empty()` (praised are E2E tests😄)